### ISSUE TITLE
Datacap 9.1.7 Desktop Panels bug

### ISSUE THREAD

**Issue by emicon123** [ibm-ecm/datacap-developer-kit#1]
Hi,

I use Datacap 9.1.7.0182. I wanted to develop a custom DCDesktop Scan Panel (ISISScan).

I have upgraded .NET to 4.7.2, target my solution to 4.7.2 and targeted newest AssemblyVersion of Datacap components (9.1.7).

After deploying and trying to open my custom panel I got error "Exception was thrown by the target of an invocation".

I have noticed that the same problem occurs for the original version of DotScanPanels.dll included in DCDesktop folder. (I've changed the custom scan panel in TravelDocs from VScan to ISISScan).

I hope you can help me to solve the problem.

**Comment by rohitzagadeibm** [comment]
Hi emicon123

Can you please raise Datacap salesforce case for this issue. So i can help you to resolve your issue. I may require to review your setup.

Below is the link from you can raise case. For that you need to login with ibm id. If you dont have that please get in touch with IBM Sales representative to file case.

https://idaas.iam.ibm.com/idaas/mtfim/sps/authsvc?PolicyId=urn:ibm:security:authentication:asf:basicldapuser

**Comment by emicon123** [comment]
I've raised a PMR already. I actually resolved this issue by installing Visual Studio 2013 on target enviroment (it seems like VS installs some required libraries or something). So in my opinion it's required to install VS in order to use custom panel on given enviroment. Please check if panels are working on enviroment with no Visual Studio 2013 installed.

**Comment by emicon123** [comment]
Here is the PMR number TS003580943.

**Comment by rohitzagadeibm** [comment]
Hi Wojciech
 
There are two different scenarios here.
 
 1. When you require Visual studio installed?
 2. When you do not require Visual Studio?
 
1.When you require Visual studio installed?
     As datacap developer, If you want develop your own custom panels at your end then you need to install visual studio on system.
 
 2. When you do not require Visual Studio?
     When user is using datacap application on their system then  they dont required visual studio installed on system. They just need to install datacap setup and if required any microsoft frameworks. Thats it and they can           use datacap.
 
Let me know if anymore help required in this.
 
Regards
RZ
 
----- Original message -----From: Wojciech Szczyrba <notifications@github.com>To: ibm-ecm/datacap-developer-kit <datacap-developer-kit@noreply.github.com>Cc: rohitzagadeibm <rozagade@in.ibm.com>, Manual <manual@noreply.github.com>Subject: [EXTERNAL] Re: [ibm-ecm/datacap-developer-kit] Datacap 9.1.7 Desktop Panels bug (#1)Date: Fri, Apr 10, 2020 6:22 PM 
 
I've raised a PMR already. I actually resolved this issue by installing Visual Studio 2013 on target enviroment (it seems like VS installs some required libraries or something). So in my opinion it's required to install VS in order to use custom panel on given enviroment. Please check if panels are working on enviroment with no Visual Studio 2013 installed.

**Comment by emicon123** [comment]
2. When you do not require Visual Studio?

That's the thing. I have installed and configured Datacap on user's system. I have also installed .NET Framework 4.8 and after changing panel in Travel docs from VScan to ISISScan it throws an error.
ISIS and TWAIN panels worked for me only after I've installed VS2013 on user's system.

Regards,
Wojciech Szczyrba

**Comment by rohitzagadeibm** [comment]
Try by just installing framework 4.6.1. Till 916 datacap releases we are supporting framework 4.6.1. After that from Datacap 917 we are supporting framework 4.7.1. I hope you are using datacap version before 917.
 
And instead of TravelDocs can you please try APT app by changing panel to ISISScan and check.
 
Regards
RZ
 
----- Original message -----From: Wojciech Szczyrba <notifications@github.com>To: ibm-ecm/datacap-developer-kit <datacap-developer-kit@noreply.github.com>Cc: rohitzagadeibm <rozagade@in.ibm.com>, Manual <manual@noreply.github.com>Subject: [EXTERNAL] Re: [ibm-ecm/datacap-developer-kit] Datacap 9.1.7 Desktop Panels bug (#1)Date: Tue, Apr 14, 2020 4:09 PM 
 
        When you do not require Visual Studio?
That's the thing. I have installed and configured Datacap on user's system. I have also installed .NET Framework 4.8 and after changing panel in Travel docs from VScan to ISISScan it throws an error.ISIS and TWAIN panels worked for me only after I've installed VS2013 on user's system.
Regards,Wojciech Szczyrba

**Comment by emicon123** [comment]
Why do you hope I am using Datacap version before 9.1.7? As I mentioned in the issue title and in my original message I use Datacap 9.1.7. That is all the case, because I know that custom panels work for version 9.1.6 and 9.1.5.

Regards,
Wojciech Szczyrba

**Comment by rohitzagadeibm** [comment]
I see you are using Datacap version  9.1.7.182 and you are applying custom panel toolkit as  DCDesktop-9.1.5-ifix5-Panels-with-Universal solution and Datacap-DDK. Please confirm if my understanding is wrong.
 
If this is the case then i will provide you the toolkit which is compatible for 9.1.7.182. The Custom panel toolkit which you are applying will not work with version Datacap 9.1.7.182.
 
Regards
RZ
 
----- Original message -----From: Wojciech Szczyrba <notifications@github.com>To: ibm-ecm/datacap-developer-kit <datacap-developer-kit@noreply.github.com>Cc: rohitzagadeibm <rozagade@in.ibm.com>, Manual <manual@noreply.github.com>Subject: [EXTERNAL] Re: [ibm-ecm/datacap-developer-kit] Datacap 9.1.7 Desktop Panels bug (#1)Date: Tue, Apr 14, 2020 4:40 PM 
 
Why do you hope I am using Datacap version before 9.1.7? As I mentioned in the issue title and in my original message I use Datacap 9.1.7. That is all the case, because I know that custom panels work for version 9.1.6 and 9.1.5.
Regards,Wojciech Szczyrba

**Comment by emicon123** [comment]
Yes, you understand properly. I try to develop my panel as DCDesktop-9.1.5-ifix5-Panels-with-Universal solution and Datacap-DDK.

Please provide my the toolkit compatible with Datacap 9.1.7.182 version.

Thank you in advance.

Regards,
Wojciech Szczyrba

**Comment by rohitzagadeibm** [comment]
Sure. I will handover this toolkit to support engineer. He will provide you that in PMR.
 
Regards
RZ
 
----- Original message -----From: Wojciech Szczyrba <notifications@github.com>To: ibm-ecm/datacap-developer-kit <datacap-developer-kit@noreply.github.com>Cc: rohitzagadeibm <rozagade@in.ibm.com>, Manual <manual@noreply.github.com>Subject: [EXTERNAL] Re: [ibm-ecm/datacap-developer-kit] Datacap 9.1.7 Desktop Panels bug (#1)Date: Tue, Apr 14, 2020 5:26 PM 
 
Yes, you understand properly. I try to develop my panel as DCDesktop-9.1.5-ifix5-Panels-with-Universal solution and Datacap-DDK.
Please provide my the toolkit compatible with Datacap 9.1.7.182 version.
Thank you in advance.
Regards,Wojciech Szczyrba
